pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -126,13 +126,24 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 		namespace string
 	}
 	{{ lower_camel .Name }}Chan := make(chan {{ lower_camel .Name }}ListWithNamespace)
-{{- end }}
+
+	var initial{{ upper_camel .Name }}List {{ .ImportPrefix }}{{ .Name }}List{{- end }}
+
 {{- end}}
 
+	currentSnapshot := {{ .GoName }}Snapshot{}
+
 	for _, namespace := range watchNamespaces {
 {{- range .Resources}}
 {{- if (not .ClusterScoped) }}
 		/* Setup namespaced watch for {{ .Name }} */
+		{
+			upstreams, err := c.{{ lower_camel .Name }}.List(namespace, clients.ListOpts{Ctx: opts.Ctx, Selector: opts.Selector})
```

Comment:
`upstreams` -> `{{ lower_camel .PluralName }}`

Reply:
no need as they are in an internal scope